### PR TITLE
session, config, sessionctx, executor: Set tidb_opt_projection_push_down to false when tidb upgrades from versions older than 8.3.0 to 8.3.0 and later

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: ref #51876 #53831

Problem Summary:

### What changed and how does it work?
In #53831, we change tidb_opt_projection_push_down session variable's default value to 'on'. After discussion with PM, we need to keep consistency with older versions and avoid any unknown potential problems, we decide to keep the 'off' value for tidb cluster upgradation. Thus we change the variable to both session and global scope. Besides, this PR deprecates the previous config item "projection-push-down", we can do this because this config would have never been used by customers due to #34993 issue. 
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by windtalker — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

File: pkg/sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -211,7 +211,7 @@ var defaultSysVars = []*SysVar{
 		}
 		return nil
 	}},
-	{Scope: ScopeSession, Name: TiDBOptProjectionPushDown, Value: BoolToOnOff(config.GetGlobalConfig().Performance.ProjectionPushDown), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBOptProjectionPushDown, Value: BoolToOnOff(DefOptEnableProjectionPushDown), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
```
Comment (hawkingrei) [inline pkg/sessionctx/variable/sysvar.go:214]: Why do we need to change this? It appears that the old behavior has changed.

Comment (yibin87) [reply pkg/sessionctx/variable/sysvar.go:214]: To allow different init values for new deployed 8.3.0 tidb cluster and upgraded 8.3.0 tidb cluster, need to change to global session scope. Besides, it makes sense to be a global session scope system variable also.

**Review by winoros — APPROVED** [review]

**Pull request merged**